### PR TITLE
Hw 4

### PR DESCRIPTION
**Pull request by AlenaKharytonchyk**
Everything is done according to the criteria.
All new components are implemented. Dialog component uses [Portals] functionality and [focus-trap-react] package.
Component tests are written to cover new components functionality. All tests pass.
All new components have working Storybook stories that showcase use cases.
Storybook stories exist that compose Dialog and MovieForm components and showcase all use cases mentioned in the description.

### CODE REVIEW

File: src/components/dialog/Dialog.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import {createPortal} from "react-dom";
+import FocusTrap from "focus-trap-react";
+
+const Dialog = ({title, children, showModal}) => {
+    const [modalVisible, setModalVisible] = useState(showModal);
+    useEffect(() => {
+        setModalVisible(showModal);
+    }, [showModal])
+
+    return modalVisible
```
Comment (AliaksandrZahorski) [inline src/components/dialog/Dialog.js:12]: what changes if we rewrite this line to `return showModal` ?

File: src/components/dialog/Dialog.scss [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  padding: 30px;
+  color: white;
+  display: flex;
+  flex-direction: column;
+  gap: 20px;
+  margin: 0 auto;
+  font-family: Montserrat, sans-serif;
+  box-shadow: 0 0 1px 0 rgba(51, 51, 51, 0.14), 0 1px 7px 0 rgba(51, 51, 51, 0.2);
+
+  &.hide {
```
Comment (AliaksandrZahorski) [inline src/components/dialog/Dialog.scss]: where you use this class?

File: src/components/dialog/Dialog.scss [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    display: none;
+  }
+
+  .dialog-body {
+    padding: 0 60px;
+  }
+
+  .close-button {
+    position: absolute;
```
Comment (AliaksandrZahorski) [inline src/components/dialog/Dialog.scss:20]: not a big fan of this way of positioning

File: src/components/movieForm/MovieForm.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        genre: 'genre',
+        runtime: 'runtime',
+        overview: 'overview'
+    };
+
+    const genreOptions = ["all", "comedy", "drama", "detective"];
+
+    Object.keys(formLabels).forEach((key) => {
+        formLabels[key] = formLabels[key].toUpperCase();
+    });
```
Comment (AliaksandrZahorski) [inline src/components/movieForm/MovieForm.js]: From line 6 to this line, variables will be created every time when the props are updated. Any constants outside the component?

File: src/components/movieTile/MovieTile.js [hunk 5]
```diff
@@ -14,7 +14,7 @@ const MovieTile = ({title, image, genre, year, handleClick}) => {
             <img src={image} alt={title} />
             <div className="title-container">
                 <h2 className="title">{title}</h2>
-                <span className="year">{year}</span>
+                <span className="year">{new Date(year).getFullYear()}</span>
```
Comment (AliaksandrZahorski) [inline src/components/movieTile/MovieTile.js]: looks like candidate for utils.js, DRY

File: src/components/dialog/Dialog.scss (hunk 2 above)
Comment (AlenaKharytonchyk) [reply src/components/dialog/Dialog.scss]: For now, nowhere. Removed. Thank you

File: src/components/movieForm/MovieForm.js (hunk 4 above)
Comment (AlenaKharytonchyk) [reply src/components/movieForm/MovieForm.js]: Thank you! Done

File: src/components/dialog/Dialog.js (hunk 1 above)
Comment (AlenaKharytonchyk) [reply src/components/dialog/Dialog.js:12]: Visibility is controlled, inside via modalVisible, and outside by showModal. However, modalVisible is initialized with show modal.

File: src/components/movieTile/MovieTile.js (hunk 5 above)
Comment (AlenaKharytonchyk) [reply src/components/movieTile/MovieTile.js]: Done

**Comment by AliaksandrZahorski** [comment]
Approved

**Pull request merged**